test/relations.test.js

```diff
@@ -819,6 +819,18 @@ describe('relations', function() {
             done();
           });
         });
+        it('error case: filtering not done with wrong id key', function(done) {
+          var wrongWhereFilter = {where: {wrongIdKey: samplePatientId}};
+          physician.patients(wrongWhereFilter, function(err, ch) {
+            if (err) return done(err);
+            should.exist(ch);
+            ch.should.have.lengthOf(3);
```

Comment:
It seems to me that when nothing matches the filter condition, it should return empty result.  According to @jannyHou, juggler or some connectors act differently.  Adding her to review.